client/van_serviceinterface_update.go

```diff
@@ -0,0 +1,355 @@
+		}
+	} else if errors.IsNotFound(err) {
+		return nil, fmt.Errorf("StatefulSet %s not found", targetName)
+	} else {
+		return nil, fmt.Errorf("Could not read StatefulSet %s: %s", targetName, err)
+	}
+}
+
+func removeServiceInterfaceTarget(serviceName string, targetName string, deleteIfNoTargets bool, cli *VanClient) error {
+	current, err := cli.KubeClient.CoreV1().ConfigMaps(cli.Namespace).Get("skupper-services", metav1.GetOptions{})
+	if err == nil {
+		jsonDef := current.Data[serviceName]
+		if jsonDef == "" {
+			return fmt.Errorf("Could not find entry for service interface %s", serviceName)
+		} else {
+			service := types.ServiceInterface{}
+			err = jsonencoding.Unmarshal([]byte(jsonDef), &service)
+			if err != nil {
+				return fmt.Errorf("Failed to read json for service interface %s: %s", serviceName, err)
+			} else {
+				modified := false
+				targets := []types.ServiceInterfaceTarget{}
+				for _, t := range service.Targets {
+					if t.Name == targetName || (t.Name == "" && targetName == serviceName) {
+						modified = true
+					} else {
+						targets = append(targets, t)
+					}
+				}
+				if !modified {
+					return fmt.Errorf("Could not find target %s for service interface %s", targetName, serviceName)
+				}
+				if len(targets) == 0 && deleteIfNoTargets {
+					delete(current.Data, serviceName)
+				} else {
+					service.Targets = targets
+					encoded, err := jsonencoding.Marshal(service)
+					if err != nil {
+						return fmt.Errorf("Failed to create json for service interface: %s", err)
+					} else {
+						current.Data[serviceName] = string(encoded)
+					}
+				}
+			}
+		}
+		_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.Namespace).Update(current)
+		if err != nil {
+			return fmt.Errorf("Failed to update skupper-services config map: ", err.Error())
+		}
+	} else if errors.IsNotFound(err) {
+		return fmt.Errorf("No skupper service interfaces defined: ", err.Error())
+	} else {
+		return fmt.Errorf("Could not retrieve service interfaces from configmap 'skupper-services'", err)
+	}
+	return nil
+}
+
+func (cli *VanClient) VanServiceInterfaceUnbind(ctx context.Context, targetType string, targetName string, address string, deleteIfNoTargets bool) error {
+	if targetType == "deployment" || targetType == "statefulset" {
+		if address == "" {
+			err := removeServiceInterfaceTarget(targetName, targetName, deleteIfNoTargets, cli)
+			return err
+		} else {
+			err := removeServiceInterfaceTarget(address, targetName, deleteIfNoTargets, cli)
+			return err
+		}
+	} else if targetType == "pods" {
+		return fmt.Errorf("Target type for service interface not yet implemented")
+	} else {
+		return fmt.Errorf("Unsupported target type for service itnerface", targetType)
+	}
+}
+
+func (cli *VanClient) VanServiceInterfaceExpose(ctx context.Context, targetType string, targetName string, options types.VanServiceInterfaceCreateOptions) error {
```

Comment:
Would prefer not to have two ways to accomplish the same operation in the interface. Is there a functional difference between expose/unexpose and bind/unbind? Could we keep expose/unexpose but somehow indicate recommended use is bind/unbind and then deprecate expose down the road?

Reply:
Agree on not having two so similar methods in the API. I have removed the expose method and now that is merely a convenience offered by the cli tool.

It might even be worth removing Bind/Unbind and just having Update.

I do still have the somewhat ugly GetHeadlessServiceConfiguration in the api. Not sure it is in quite the right shape or place yet but perhaps we can live with it for now and discuss some more to see if we can find a nicer way to handle that.